### ISSUE TITLE
interface improvements?

### ISSUE THREAD

**Issue by ghost** [pyrovski/wrtbwmon#7]
hey, great to know you have continued on wrtbwmon script. I really don't know why something like this isn't built in OpenWRT! really!

1) I've added IP column next to User column following your code. I didn't find any reason to not showing it. I haven't taken a look at the "awk" branch although, so I stopped from pulling a request to master on that little change.

2) are you planning to improve the design a bit? bootstrap? maybe integrate with luci interface as luci-app-wrtbwmon?

3) if I do `wrtbwmon publish /tmp/usage.db /tmp/usage.htm` every minute instead of an hour inside my cronjob, would it be resource consuming? would it be possible to only publish when accessing the html file? and maybe include an autorefresh feature?

sorry, lot of questions :)

**Comment by pyrovski** [comment]
Thanks! OpenWRT has decent real-time bandwidth monitoring, but they are very conscientious about adding background processes or cron jobs. I haven't taken this up with project yet, but I feel that having wrtbwmon around can solve the problem for many people.

1: I'll make the "awk" branch the default; the code is much cleaner.

2: I don't think I have time for this. If you want to do it or know someone who does, go right ahead.

3: That's probably fine, but depends on your router CPU and where you are storing the database and html file. I have a branch somewhere with autorefresh and per-host charts that I will upload soon.

**Comment by ghost** [comment]
thanks for your reply
I hope you could upload that branch soon :)

**Comment by pyrovski** [comment]
Check out the "experimental" branch. The wrtbwmon.sh functionality should be the same, but there's also a continuous.sh script that creates an in-memory time-series database of bandwidth measurements per host, and relays this to the user through CGI/JS. It has serious issues with the time-series data compaction, so don't trust the numbers yet.

**Comment by Kiougar** [comment]
Hey @pyrovski 

>2. are you planning to improve the design a bit? bootstrap? maybe integrate with luci interface as luci-app-wrtbwmon?

I created the [luci-wrtbwmon](https://github.com/Kiougar/luci-wrtbwmon) which is a Luci Web UI module that uses `wrtbwmon`; it also has some additional features.

But I kinda need help creating an `.ipk` for it. I would appreciate it if you could help me a little bit on that part.

Thanks in advance!

**Comment by pyrovski** [comment]
IPKs are similar to Debian packages. If you can find a copy of the ipkg-build script somewhere, the rest boils down to placing the input files in a particular directory layout. [Here's an overview](https://thingshardtofind.blogspot.com/2015/03/ipkg-howto.html) (I haven't followed it myself). I'll take a look at your project when I have time. Thanks for contributing!

**Comment by vbsampath** [comment]
Hey guys
I used this package for a while now and I changed it and integrated to luci web interface.
I would like to integrate my work into the main package if possible. Below are screenshots of my work.
The last one is after sorting the total column. I would like to contribute to this package in any way possible.

![1](https://user-images.githubusercontent.com/7717661/28274439-8fa00318-6b2e-11e7-9406-983b370eecbb.JPG)
![2](https://user-images.githubusercontent.com/7717661/28274441-8fa7cfb2-6b2e-11e7-9d4b-6dda660d7328.JPG)
![3](https://user-images.githubusercontent.com/7717661/28274440-8fa7e3c6-6b2e-11e7-85e5-cf50d5ddc0bb.JPG)
